### ISSUE TITLE
Question regarding train / val / test split

### ISSUE THREAD

**Issue by victor714** [allenai/specter#9]
Please help me with following queries on train / val / test dataset split

1. What was the basis used for splitting corpus into train, val and test? 

2. In the training script provided (as below), is there option to include `test-path` so that the best model saved based on results on val can be evaluated on the test set?
```
./scripts/run-exp-simple.sh -c experiment_configs/simple.jsonnet \
-s [output-dir] --num-epochs [num-epochs] --batch-size [batch-size] \
--train-path [path-to-train.pkl] --dev-path [path-to-dev.pkl] \
--cuda-device 0 --num-train-instances [num-instances]
```

**Comment by armancohan** [comment]
1- The only metric that is computed during training is loss and since loss is not very useful for evaluating the effectiveness of the embeddings, we do not use a final test set for this. In initial version we planned to implement additional metrics in the main training script but as the evaluation tasks grew we decided to have a separate full benchmark for evaluating the final embeddings. We encourage using a dev set for tracking loss but do final evaluation using the full SciDocs benchmark: https://github.com/allenai/scidocs

2- If for your specific use case you need to pass in a separate test set, you need to modify the jsonnet config file: Uncomment lines [7](https://github.com/allenai/specter/blob/541fc2bf8b95118665082a335737508b3ce2c9b8/experiment_configs/simple.jsonnet#L7) and [143](https://github.com/allenai/specter/blob/541fc2bf8b95118665082a335737508b3ce2c9b8/experiment_configs/simple.jsonnet#L143) and add a `evaluate_on_test=true` under the `trainer` config in line [144](https://github.com/allenai/specter/blob/541fc2bf8b95118665082a335737508b3ce2c9b8/experiment_configs/simple.jsonnet#L144)

**Comment by victor714** [comment]
Thanks @armancohan and how the publication ids in corpus were assigned to train/val/test? Was it done randomly or based on some other metric?

**Comment by armancohan** [comment]
We randomly split it. If you provide these ids to the create_training_files script it will generate the corresponding instances in each set for you. details in [readme](https://github.com/allenai/specter#advanced-training-your-own-model):
